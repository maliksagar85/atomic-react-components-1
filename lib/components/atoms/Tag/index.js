export { default } from './Tag';
export { TagVanilla } from './Tag';
