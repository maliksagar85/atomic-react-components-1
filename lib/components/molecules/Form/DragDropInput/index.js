export { default } from './DragDropInput';
export { DDFieldInputVanilla } from './DragDropInput';
