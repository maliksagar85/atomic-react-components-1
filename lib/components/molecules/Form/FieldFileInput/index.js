export { default } from './FieldFileInput';
export { FieldFileInputVanilla } from './FieldFileInput';
