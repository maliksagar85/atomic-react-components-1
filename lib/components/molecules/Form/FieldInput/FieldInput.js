import React from 'react';
import PropTypes from 'prop-types';
import uniqueId from 'lodash/uniqueId';
import styled from 'styled-components';
import Input from '../../../atoms/Input';
import FieldError from '../../../atoms/FieldError';
import Label from '../../../atoms/Label';
import styles from './FieldInput.style';
import Para from '../../../atoms/Para';

const FieldInput = ({
  id,
  className,
  type,
  name,
  label,
  onChange,
  onBlur,
  labelProps,
  fieldProps,
  validationInstructions,
  register,
  errors,
  ...others
}) => {
  const inputID = id || uniqueId('input-');
  return (
    // TODO: Add support for label position
    <div className={className}>
      <Label htmlFor={id} {...labelProps}>
        {label}
      </Label>
      <Input
        type={type}
        name={name}
        {...others}
        aria-describedby={validationInstructions ? `${inputID}-instruction` : null}
        ref={register}
      />
      {validationInstructions ? (
        <Para id={`${inputID}-instruction`}>{validationInstructions}</Para>
      ) : (
        ''
      )}
      {errors[name] && <FieldError>{errors[name].message}</FieldError>}
    </div>
  );
};

FieldInput.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.node.isRequired,
  labelProps: PropTypes.objectOf(PropTypes.object),
  fieldProps: PropTypes.objectOf(PropTypes.object),
  placeholder: PropTypes.string,
  validationInstructions: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  register: PropTypes.func.isRequired,
  errors: PropTypes.objectOf(PropTypes.object),
};

FieldInput.defaultProps = {
  placeholder: '',
  validationInstructions: '',
  className: '',
  type: 'text',
  labelProps: {},
  fieldProps: {},
  onChange: null,
  onBlur: null,
  errors: {},
};

const StyledFieldInput = styled(FieldInput)`
  ${styles};
`;

export default StyledFieldInput;
export { FieldInput as FieldInputVanilla };
