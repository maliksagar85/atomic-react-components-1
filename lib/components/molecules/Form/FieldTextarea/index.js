export { default, FieldTextareaVanilla } from './FieldTextarea';
