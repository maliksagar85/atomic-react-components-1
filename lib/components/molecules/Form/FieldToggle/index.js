export { default } from './FieldToggle';
export { FieldToggleVanilla } from './FieldToggle';
