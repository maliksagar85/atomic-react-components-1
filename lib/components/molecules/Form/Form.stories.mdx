import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import {
  userFormProps,
  userFormSelectProps,
  fileMultipleInputProps,
  fileSingleInputProps,
  fileDDInputProps,
  userFormNameInputProps,
  userFormEmailProps,
  userFormCommentProps,
  userFormAgeGroupProps,
  userFormTnCProps,
  userFormToggleProps,
} from './Form.mock';
import Form from './Form'
import Button from '../../atoms/Button';

<Meta title="Molecules/Form" component={Form} />

### Default

<Preview>
  <Story name="Default">
    <Form {...userFormProps}>
      <Form.Select {...userFormSelectProps} />
      <Form.Input {...userFormNameInputProps} />
      <Form.Input {...userFormEmailProps} />
      <Form.RadioGroup {...userFormAgeGroupProps} />
      <Form.Checkbox {...userFormTnCProps} />
      <Form.Textarea {...userFormCommentProps} />
      <Form.Toggle {...userFormToggleProps} />
      <Form.FieldFile {...fileSingleInputProps} />
      <Form.FieldFile {...fileMultipleInputProps} />
      <Form.DDFile {...fileDDInputProps} />
      <Button primary type="submit">
        Submit
      </Button>
    </Form>
  </Story>
</Preview>

## Props ðŸ”§

<Props of={Form} />


