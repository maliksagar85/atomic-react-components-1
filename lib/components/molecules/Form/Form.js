import React from 'react';
import { useForm } from 'react-hook-form';
import PropTypes from 'prop-types';

import { buildSchema } from '../../../core/utils/buildValidationSchema';
import Debug from './Debug';
import FieldInput from './FieldInput';
import FieldSelect from './FieldSelect';
import FieldTextarea from './FieldTextarea';
import FieldRadioGroup from './FieldRadioGroup';
import FieldCheckbox from './FieldCheckbox';
import FieldToggle from './FieldToggle';
import FieldFileInput from './FieldFileInput';
import DDFileInput from './DragDropInput';

const Form = ({
  onSubmit,
  initialValues,
  children,
  debug,
  validate,
  validationSchema,
  yupValidationSchema,
}) => {
  const getSchema = () => {
    if (yupValidationSchema) return yupValidationSchema;
    return validationSchema ? buildSchema(validationSchema) : null;
  };
  const { handleSubmit, register, errors, setValue } = useForm({
    mode: validate,
    validationSchema: getSchema(yupValidationSchema, validationSchema),
    defaultValues: initialValues,
  });

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {React.Children.map(children, (child) =>
        React.cloneElement(child, { register, errors, setValue })
      )}
      {debug && <Debug />}
    </form>
  );
};

Form.Checkbox = FieldCheckbox;

Form.Input = FieldInput;

Form.RadioGroup = FieldRadioGroup;

Form.Select = FieldSelect;

Form.Textarea = FieldTextarea;

Form.Toggle = FieldToggle;

Form.FieldFile = FieldFileInput;

Form.DDFile = DDFileInput;

Form.defaultProps = {
  debug: false,
  initialValues: null,
  validationSchema: null,
  yupValidationSchema: null,
  validate: 'onSubmit',
};

Form.propTypes = {
  initialValues: PropTypes.objectOf(PropTypes.object),
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func.isRequired,
  debug: PropTypes.bool,
  validationSchema: PropTypes.objectOf(PropTypes.object),
  validate: PropTypes.string,
  yupValidationSchema: PropTypes.func,
};

export default Form;
