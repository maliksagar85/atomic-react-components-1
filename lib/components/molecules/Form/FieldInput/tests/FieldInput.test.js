import React from 'react';

import { shallowWithTheme, renderWithTheme } from '../../../../../../__mock__/themeMock';
import Form from '../../Form';
import { FieldInputVanilla } from '../FieldInput';
import { ButtonVanilla } from '../../../../atoms/Button';

describe('FieldInput Component', () => {
  let FieldInput = '';

  test('should render correctly', () => {
    const props = {
      label: 'Enter username',
      name: 'username',
      type: 'text',
      validationInstructions: 'Enter valid username',
    };
    FieldInput = shallowWithTheme(
      <Form>
        <div>
          <FieldInputVanilla {...props} />
          <FieldInputVanilla {...props} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldInput).toMatchSnapshot();
  });

  test('should render label correctly', () => {
    const props = {
      label: 'Enter username',
      name: 'username',
      type: 'text',
      validationInstructions: 'Enter valid username',
    };
    FieldInput = renderWithTheme(
      <Form>
        <div>
          <FieldInputVanilla {...props} />
          <FieldInputVanilla {...props} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    console.log('TCL: FieldInput', FieldInput);

    expect(FieldInput.find('label').text()).toEqual('Enter usernameEnter username');
  });

  test('should render Input component with proper type', () => {
    const props = {
      name: 'email',
      type: 'email',
      id: 'email',
    };
    FieldInput = renderWithTheme(
      <Form>
        <div>
          <FieldInputVanilla {...props} />
          <FieldInputVanilla {...props} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldInput.find('input')).toHaveLength(2);
  });
});
