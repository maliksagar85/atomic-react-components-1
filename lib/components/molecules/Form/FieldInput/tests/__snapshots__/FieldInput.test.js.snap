// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput Component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    errors={Object {}}
    key=".0"
    register={[Function]}
    setValue={[Function]}
  >
    <FieldInput
      className=""
      errors={Object {}}
      fieldProps={Object {}}
      label="Enter username"
      labelProps={Object {}}
      name="username"
      onBlur={null}
      onChange={null}
      placeholder=""
      type="text"
      validationInstructions="Enter valid username"
    />
    <FieldInput
      className=""
      errors={Object {}}
      fieldProps={Object {}}
      label="Enter username"
      labelProps={Object {}}
      name="username"
      onBlur={null}
      onChange={null}
      placeholder=""
      type="text"
      validationInstructions="Enter valid username"
    />
  </div>
  <Button
    ariaLabel=""
    className=""
    disabled={false}
    elementLocator=""
    errors={Object {}}
    inheritedStyles=""
    key=".1"
    primary={true}
    register={[Function]}
    secondary={false}
    setValue={[Function]}
    tertiary={false}
    transparent={false}
    type="submit"
    variation=""
  >
    Submit
  </Button>
</form>
`;
